src/store/reducers/userAuthReducer.ts

```diff
@@ -0,0 +1,104 @@
+import { IUser } from "../../models/request/IUser";
+import {
+  UserActionTypes,
+  UserAuthAction,
+  UserAuthState,
+} from "../../pages/types/User";
+
+const initialUserState: UserAuthState = {
+  isLoggedIn: false,
+  loading: false,
+  user: {} as IUser,
+  accessToken: "",
+  refreshToken: "",
+  message: "",
+};
+
+export const userAuthReducer = (
+  state = initialUserState,
+  action: UserAuthAction
+): UserAuthState => {
+  switch (action.type) {
+    case UserActionTypes.LOGIN_ACTION:
+      return {
+        isLoggedIn: false,
+        loading: true,
+        message: "Login...",
+      };
+    case UserActionTypes.LOGIN_FAILED:
+      return {
```

Comment:
redux is unmutable
you should return the same state with changed fields
return {
...state,
isLoading: false,
loading: true
}

read about this